domain/src/test/java/org/oppia/domain/oppialogger/analytics/AnalyticsControllerTest.kt

```diff
@@ -358,13 +399,108 @@ class AnalyticsControllerTest {
     assertThat(eventContext.revisionCardContext.subTopicId).matches(TEST_SUB_TOPIC_ID)
   }
 
+  @ExperimentalCoroutinesApi
+  @InternalCoroutinesApi
+  @Test
+  fun testController_logTransitionEvent_withNoNetwork_checkLogsEventToStore() {
+    networkConnectionUtil.setCurrentConnectionStatus(NONE)
+    analyticsController.logTransitionEvent(
+      TEST_TIMESTAMP,
+      EventAction.EVENT_ACTION_UNSPECIFIED,
+      analyticsController.createQuestionContext(
+        TEST_QUESTION_ID,
+        listOf(
+          TEST_SKILL_LIST_ID, TEST_SKILL_LIST_ID
+        )
+      )
+    )
+
+    val eventLogs = analyticsController.getEventLogs()
+    eventLogs.observeForever(this.mockOppiaEventLogsObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+    verify(
+      this.mockOppiaEventLogsObserver,
+      atLeastOnce()
+    ).onChanged(oppiaEventLogsResultCaptor.capture())
+
+    val eventLog = oppiaEventLogsResultCaptor.value.getOrThrow().getEventLog(0)
+    assertThat(eventLog.priority).isEqualTo(Priority.ESSENTIAL)
+    assertThat(eventLog.context.activityContextCase).isEqualTo(QUESTION_CONTEXT)
+    assertThat(eventLog.timestamp).isEqualTo(TEST_TIMESTAMP)
+    assertThat(eventLog.actionName).isEqualTo(EventAction.EVENT_ACTION_UNSPECIFIED)
+  }
+
+  @ExperimentalCoroutinesApi
+  @InternalCoroutinesApi
+  @Test
+  fun testController_logClickEvent_withNoNetwork_checkLogsEventToStore() {
+    networkConnectionUtil.setCurrentConnectionStatus(NONE)
+    analyticsController.logClickEvent(
+      TEST_TIMESTAMP,
+      EventAction.EVENT_ACTION_UNSPECIFIED,
+      analyticsController.createQuestionContext(
+        TEST_QUESTION_ID,
+        listOf(
+          TEST_SKILL_LIST_ID, TEST_SKILL_LIST_ID
+        )
+      )
+    )
+
+    val eventLogs = analyticsController.getEventLogs()
+    eventLogs.observeForever(this.mockOppiaEventLogsObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+    verify(
+      this.mockOppiaEventLogsObserver,
+      atLeastOnce()
+    ).onChanged(oppiaEventLogsResultCaptor.capture())
+
+    val eventLog = oppiaEventLogsResultCaptor.value.getOrThrow().getEventLog(0)
```

Comment:
Do we verify that this is a click vs. a transition event is logged & cached for this & the test above? That seems like an important distinction to verify.

Reply:
Yes, we do that by checking the priority of the logged event. If it's `OPTIONAL` then it points to a click event and if it's `ESSENTIAL` then it points to a transition event.

Reply:
Maybe its worth specifying that in a comment here

Reply:
(Sarthak you can resolve this after adding a comment)

Reply:
Added comments. Resolving.